Sources/PointFree/Account/Account.swift

```diff
@@ -149,32 +149,125 @@ private func planName(for subscription: Stripe.Subscription) -> String {
-    gridColumn(sizes: [.mobile: 2], [
-      div([
-        "Plan"
+public func status(for subscription: Stripe.Subscription) -> String {
+  switch subscription.status {
+  case .active:
+    return "Active"
+  case .canceled:
+    return "Canceled"
+  case .pastDue:
+    return "Past due"
+  case .unpaid:
+    return "Unpaid"
+  case .trialing:
+    return "In trial"
+  }
+}
+
+public func nextBilling(for subscription: Stripe.Subscription) -> String {
+  switch subscription.status {
+  case .active:
+    let totalAmountString = totalAmount(for: subscription) ?? ""
+    let currentPeriodEndString = subscription.currentPeriodEnd
+      .map { " on " + dateFormatter.string(from: $0) } ?? ""
+    return totalAmountString + currentPeriodEndString
+  case .canceled:
+    return subscription.currentPeriodEnd
+      .filterOptional { $0 > AppEnvironment.current.date() }
+      .map { "Cancels " + dateFormatter.string(from: $0) }
+      ?? "Canceled"
+  case .pastDue:
+    return "" // FIXME
+  case .unpaid:
+    return "" // FIXME
+  case .trialing:
+    return "" // FIXME
+  }
+}
+
+private let subscriptionPlanRows = View<Stripe.Subscription> { subscription in
+  return div([`class`([Class.padding([.mobile: [.top: 1, .bottom: 3]])])], [
+    gridRow([
+      gridColumn(sizes: [.mobile: 3], [
+        p([div(["Plan"])])
+        ]),
+      gridColumn(sizes: [.mobile: 9], [
+        gridRow([
+          gridColumn(sizes: [.mobile: 12, .desktop: 6], [
+            div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
+              p([text(planName(for: subscription))])
+              ])
+            ]),
+          gridColumn(sizes: [.mobile: 12, .desktop: 6], [
+            div([`class`([Class.padding([.mobile: [.leftRight: 1]]), Class.grid.end(.desktop)])], [
+              p([
+                a([
+                  `class`([Class.pf.components.button(color: .purple, size: .small)]),
+                  href("#")
+                  ],
+                  ["Upgrade"])
+                ])
+              ])
+            ])
+          ])
         ])
       ]),
-    gridColumn(sizes: [.mobile: 10], [
-      div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
-        text(planName(for: subscription))
+    gridRow([
+      gridColumn(sizes: [.mobile: 3], [
+        p([div(["Status"])])
+        ]),
+      gridColumn(sizes: [.mobile: 9], [
+        gridRow([
+          gridColumn(sizes: [.mobile: 12, .desktop: 6], [
+            div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
+              p([text(status(for: subscription))])
+              ])
+            ]),
+          gridColumn(sizes: [.mobile: 12, .desktop: 6], [
+            div([`class`([Class.padding([.mobile: [.leftRight: 1]]), Class.grid.end(.desktop)])], [
+              p([
+                a([
+                  `class`([Class.pf.components.button(color: .purple, size: .small)]),
+                  href("#")
+                  ],
+                  ["Cancel"])
```

Comment:
I need to update our style guide for text only buttons. feel this button should prob not be accepted like all the other positive-oriented actions, and should prob be red.